pkg/splunk/enterprise/configuration.go

```diff
@@ -778,76 +783,125 @@ func AreRemoteVolumeKeysChanged(client splcommon.ControllerClient, cr splcommon.
+			_, err := checkIfVolumeExists(appFramework.VolList, appSrc.VolName)
+			if err != nil {
+				return fmt.Errorf("Invalid Volume Name for App Source: %s. %s", appSrc.Name, err)
+			}
+			vol = appSrc.VolName
+		} else {
+			if appFramework.Defaults.VolName == "" {
+				return fmt.Errorf("volumeName is missing for App Source: %s", appSrc.Name)
+			}
+			vol = appFramework.Defaults.VolName
+		}
 
-	if strings.ToLower(appFramework.Type) != "s3" {
-		return fmt.Errorf("Currently supported type for Apps Remote Storage Type is s3 only")
-	}
+		if appSrc.Scope != "" {
+			if localScope && appSrc.Scope != "local" {
+				return fmt.Errorf("Invalid scope for App Source: %s. Only local scope is supported for this kind of CR", appSrc.Name)
+			}
+
+			if appSrc.Scope != "local" && appSrc.Scope != "cluster" {
+				return fmt.Errorf("Scope for App Source: %s should be either local Or cluster", appSrc.Name)
+			}
+		}
+
+		if _, ok := duplicateChecker[vol+appSrc.Location]; ok {
+			return fmt.Errorf("Duplicate App Source configured for Volume: %s, and Location: %s combo. Remove the duplicate entry and reapply the configuration", vol, appSrc.Location)
+		}
+		duplicateChecker[vol+appSrc.Location] = true
 
-	if appFramework.S3Endpoint == "" {
-		return fmt.Errorf("Apps Remote Storage S3Endpoint is missing")
 	}
 
-	if appFramework.S3Bucket == "" {
-		return fmt.Errorf("Apps Remote Storage S3Bucket is missing")
+	if localScope && appFramework.Defaults.Scope != "" && appFramework.Defaults.Scope != "local" {
+		return fmt.Errorf("Invalid scope for defaults config. Only local scope is supported for this kind of CR")
 	}
 
-	if appFramework.S3SecretRef == "" {
-		return fmt.Errorf("Apps Remote Storage S3SecretRef is missing")
+	if appFramework.Defaults.Scope != "" && appFramework.Defaults.Scope != "local" && appFramework.Defaults.Scope != "cluster" {
+		return fmt.Errorf("Scope for defaults should be either local Or cluster, but configured as: %s", appFramework.Defaults.Scope)
 	}
 
-	if appFramework.S3PollInterval == 0 {
-		// this means either the S3PollInterval was not provided
-		// or it was explictely set to 0 - in both cases we will
-		// set the S3PollInterval to default 60 minutes
-		scopedLog.Info("Setting S3PollInterval to default 60 minutes")
-		appFramework.S3PollInterval = 60
+	if appFramework.Defaults.VolName != "" {
+		_, err := checkIfVolumeExists(appFramework.VolList, appFramework.Defaults.VolName)
+		if err != nil {
+			return fmt.Errorf("Invalid Volume Name for Defaults. Error: %s", err)
+		}
 	}
 
 	return nil
 }
 
-// ValidateSplunkSmartstoreSpec checks and validates the smartstore config
-func ValidateSplunkSmartstoreSpec(smartstore *enterprisev1.SmartStoreSpec) error {
-	var err error
+//  isAppFrameworkConfigured checks and returns true if App Framework is configured
+//  App Repo config without any App sources will not cause any App Framework activity
+func isAppFrameworkConfigured(appFramework *enterprisev1.AppFrameworkSpec) bool {
```

Comment:
This is basically our feature flag, correct?  If both `appRepo` and `appSources` are not configured, App framework is effectively disabled?

Reply:
Correct. To be more clear, the config is not effective, unless the App sources are specified.